### PR TITLE
Refactoring folders

### PR DESCRIPTION
**Pull request by PetrovKP**
image

### CODE REVIEW

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: restart
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

File: .ci/pipeline/ci.yml [hunk 1]
```diff
@@ -154,7 +167,7 @@ jobs:
 
 - job: Windows_DPCPP
   pool:
-    vmImage: 'windows-latest'
+    vmImage: 'vs2017-win2016'
```
Comment (napetrov) [inline .ci/pipeline/ci.yml:170]: Agree that it worth pining, but let's do this for latest vs and windows. 2019

File: .ci/pipeline/ci.yml [hunk 2]
```diff
@@ -107,23 +107,31 @@ jobs:
...
       conda activate CB
-      pip install daal numpy
-      pip install -r requirements-test.txt
-      conda install -y /usr/share/miniconda/envs/CB/conda-bld/linux-64/daal4py*.tar.bz2
-      python setup_sklearnex.py install --single-version-externally-managed --record=record1.txt
-      cd ..
-      pytest s/sklearnex/tests/
-    displayName: conda build
+      export DALROOT=$CONDA_PREFIX
+      ./scripts/build.sh
```
Comment (michael-smirnov) [inline .ci/pipeline/ci.yml:123]: You silently dropped the build of DPC++ components of the package. Why?

File: scripts/bld.bat [hunk 3]
```diff
@@ -16,11 +16,11 @@ rem limitations under the License.
 rem ============================================================================
 
 set DAAL4PY_VERSION=%PKG_VERSION%
-set MPIROOT=%PREFIX%\Library
+set MPIROOT=%CONDA_PREFIX%\Library
```
Comment (michael-smirnov) [inline scripts/bld.bat:19]: Why this is still a `PREFIX` for linux then?

File: scripts/run_test.bat [hunk 4]
```diff
@@ -15,26 +15,6 @@ rem See the License for the specific language governing permissions and
 rem limitations under the License.
 rem ============================================================================
 
-set DAAL4PY_VERSION=%PKG_VERSION%
-set MPIROOT=%PREFIX%\Library
-
-IF DEFINED DPCPPROOT (
-    echo "Sourcing DPCPPROOT"
-    call "%DPCPPROOT%\env\vars.bat"
```
Comment (michael-smirnov) [inline scripts/run_test.bat:23]: DPC++ RT is not needed for testing?

**Review by michael-smirnov — COMMENTED** [review]
general: I think all the files related to conda-build are better to move to a subfolder of `scripts` rather than placing it in a root. For example, they can be moved to `scripts/conda-recipe`

File: .ci/pipeline/ci.yml [hunk 5]
```diff
@@ -107,23 +107,31 @@ jobs:
...
+      conda create -q -y -n CB -c intel python=$(python.version) dal-devel impi-devel
+    displayName: 'Conda create'
   - script: |
-      export DPCPPROOT=/opt/intel/oneapi/compiler/latest
       . /usr/share/miniconda/etc/profile.d/conda.sh
       conda activate CB
-      conda build --override-channels -c intel -c conda-forge --numpy=1.19 conda-recipe
+      pip install -r requirements-dev.txt
+      pip install -r requirements-test.txt
+      pip install xgboost lightgbm
```
Comment (napetrov) [inline .ci/pipeline/ci.yml:117]: why don't have them in requirements?

**Comment by PetrovKP** [comment]
@michael-smirnov 
> general: I think all the files related to conda-build are better to move to a subfolder of scripts rather than placing it in a root. For example, they can be moved to scripts/conda-recipe

No, we want to build without a conda-build for any jobs. This scripts use for overall testing in public/private CI, without conda-build

File: scripts/bld.bat (hunk 3 above)
Comment (PetrovKP) [reply scripts/bld.bat:19]: thx, fixed

**Comment by michael-smirnov** [comment]
> This scripts use for overall testing in public/private CI, without conda-build

@PetrovKP In that case I think it worth aligning the names of build scripts on Windows (bld.bat) and Linux (build.sh) if conda-build is not used anymore. And please consider the option to place build files to a `script/build`, testing scripts to `script/test` subfolders, because the root of `script` folder looks like a mess.

**Comment by michael-smirnov** [comment]
@PetrovKP maybe it is even better to align with oneDAL repo and split `script` folder on `dev` and `deploy` scripts.

File: scripts/run_test.bat (hunk 4 above)
Comment (PetrovKP) [reply scripts/run_test.bat:23]: DPC++ RT sources by the compiler

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

File: scripts/run_test.bat (hunk 4 above)
Comment (michael-smirnov) [reply scripts/run_test.bat:23]: But you removed the line that sources the compiler environment in this change

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: restart
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=refactoring_folders

**Pull request closed without merging**